server/args_parsers_test.go

```diff
@@ -0,0 +1,128 @@
+	}
+}
+
+func TestMulticastArgsParser(t *testing.T) {
+	tt := []struct {
+		name              string
+		args              []interface{}
+		structToHidratate *MulticastArgs
+		expectedStruct    MulticastArgs
+		expectedResult    bool
+	}{
+		{name: "MuticastArgsParser Success",
+			args:              []interface{}{"sessionKey", []interface{}{int64(1000010001), int64(1000010002)}, []interface{}{"arg1_Server1", "arg1_Server2"}, []interface{}{"arg2_Server1", "arg2_Server2"}},
+			structToHidratate: &MulticastArgs{},
+			expectedStruct:    MulticastArgs{HubSessionKey: "sessionKey", ServerIDs: []int64{1000010001, 1000010002}, ServerArgs: [][]interface{}{{"arg1_Server1", "arg1_Server2"}, {"arg2_Server1", "arg2_Server2"}}},
+			expectedResult:    true},
+		{name: "MuticastArgsParser Failed",
+			args:              []interface{}{"sessionKey", []interface{}{int64(1000010001), int64(1000010002)}, []interface{}{"arg1_Server1", "arg1_Server2"}, []interface{}{"arg2_Server1", "arg2_Server2"}},
+			structToHidratate: &MulticastArgs{},
+			expectedStruct:    MulticastArgs{HubSessionKey: "", ServerIDs: []int64{1000010001, 1000010002}, ServerArgs: [][]interface{}{{"arg1_Server1", "arg1_Server2"}, {"arg2_Server1", "arg2_Server2"}}},
+			expectedResult:    false},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			new(MulticastArgsParser).Parse(tc.args, tc.structToHidratate)
+			if reflect.DeepEqual(tc.structToHidratate, &tc.expectedStruct) != tc.expectedResult {
+				t.Fatalf("expected and actual doesn't match, Expected was: %v", true)
+			}
+		})
+	}
+}
+
+func TestUnicastArgsParser(t *testing.T) {
+	tt := []struct {
+		name              string
+		args              []interface{}
+		structToHidratate *UnicastArgs
+		expectedStruct    UnicastArgs
+		expectedResult    bool
+	}{
+		{name: "UnicastArgsParser Success",
+			args:              []interface{}{"sessionKey", int64(1000010001), "arg1_Server1", "arg2_Server1"},
+			structToHidratate: &UnicastArgs{},
+			expectedStruct:    UnicastArgs{HubSessionKey: "sessionKey", ServerID: int64(1000010001), ServerArgs: []interface{}{"arg1_Server1", "arg2_Server1"}},
+			expectedResult:    true},
+		{name: "UnicastArgsParser Failed",
+			args:              []interface{}{"sessionKey", int64(1000010001), "arg1_Server1", "arg2_Server1"},
+			structToHidratate: &UnicastArgs{},
+			expectedStruct:    UnicastArgs{HubSessionKey: "", ServerID: int64(1000010001), ServerArgs: []interface{}{"arg1_Server1", "arg2_Server1"}},
+			expectedResult:    false},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			new(UnicastArgsParser).Parse(tc.args, tc.structToHidratate)
+			if reflect.DeepEqual(tc.structToHidratate, &tc.expectedStruct) != tc.expectedResult {
+				t.Fatalf("expected and actual doesn't match, Expected was: %v", true)
+			}
+		})
+	}
+}
+
+func TestListParser(t *testing.T) {
+	tt := []struct {
+		name              string
+		args              []interface{}
+		structToHidratate *ListArgs
```

Comment:
What English is this "to hidratate"? Please don't use Spanish in variables. :wink:

Reply:
Right! My bad